smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/WaiterGenerator.java

```diff
@@ -0,0 +1,223 @@
+            String expectedState = "{ state: '" + state.toString() + "'}";
+            expressionVisitor.run();
+
+            switch (pathMatcher.getComparator()) {
+                case ALL_STRING_EQUALS:
+                    expressionVisitor.writeAllStringEqualsExpectation(pathMatcher.getExpected(), expectedState);
+                    break;
+                case ANY_STRING_EQUALS:
+                    expressionVisitor.writeAnyStringEqualsExpectation(pathMatcher.getExpected(), expectedState);
+                    break;
+                case STRING_EQUALS:
+                    expressionVisitor.writeStringExpectation(pathMatcher.getExpected(), expectedState);
+                    break;
+                case BOOLEAN_EQUALS:
+                    expressionVisitor.writeBooleanExpectation(pathMatcher.getExpected(), expectedState);
+                    break;
+                default:
+                    throw new CodegenException("Invalid Matcher Comparator");
+            }
+        });
+    }
+
+    private void writeAcceptors(String accessor, boolean isException) {
+        waiter.getAcceptors().forEach((Acceptor acceptor) -> {
+            switch (acceptor.getMatcher().getMemberName()) {
+               case "success":
+                   Matcher.SuccessMember successMember = (Matcher.SuccessMember) acceptor.getMatcher();
+                   if (successMember.getValue() != isException) {
+                       generateSuccessMatcher(successMember, acceptor.getState());
+                   }
+                   break;
+               case "errorType":
+                   if (isException) {
+                       generateErrorMatcher(accessor, (Matcher.ErrorTypeMember) acceptor.getMatcher(),
+                               acceptor.getState());
+                   }
+                   break;
+               case "inputOutput":
+                   if (!isException) {
+                       Matcher.InputOutputMember member = (Matcher.InputOutputMember) acceptor.getMatcher();
+                       generatePathMatcher(accessor, member.getValue(), acceptor.getState());
+                       generatePathMatcher("input", member.getValue(), acceptor.getState());
+                   }
+                   break;
+               case "output":
+                   if (!isException) {
+                       Matcher.OutputMember member = (Matcher.OutputMember) acceptor.getMatcher();
+                       generatePathMatcher(accessor, member.getValue(), acceptor.getState());
+                   }
+                   break;
+               default:
+                   throw new CodegenException("Unknown matcher member name: " + acceptor.getMatcher().getMemberName());
+            }
+        });
+    }
+
+    private void generateAcceptors() {
+        writer.openBlock("const checkState = async (client: $L, input: $L): Promise<WaiterResult> => {", "}",
+            serviceSymbol.getName(), inputSymbol.getName(), () -> {
```

Comment:
I think using `$T` instead would work here, then you don't need to call `getName`

Reply:
Done!